### PR TITLE
add remove

### PR DESCRIPTION
**Pull request by EroAuditore**
- Implement a function for adding a new task (add a new element to the array).
- Implement a function for editing task descriptions.
- Implement a function for deleting a task (remove an element from the array).
- Implement a function for the "Clear all completed" button (use filter() method).
- The To Do List is saved in local storage.

### CODE REVIEW

**Review by MahmoudBakr23 — COMMENTED** [review]
# Hi @EroAuditore 😃,

As much as I appreciate your hard work on this project, I'm afraid that I can't perform a review on this pull request.

## **Invalid Submission**

- Kindly make sure to follow the correct linters configuration, you can refer to the this [link](https://github.com/microverseinc/linters-config/tree/master/html-css-js)  as reminder. 

_Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready._ 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._

<hr>

_Invalid Code Review Request does not count into the code reviews limit._

File: src/components/crudList.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  removeCompleted(completed);
+  saveData();
+};
+
+const deleteTask = (index) => {
+  const removed = data.filter((obj) => obj.index !== index);
+  clearData();
+  removeCompleted(removed);
+  saveData();
+};
```
Comment (voscarmv) [inline src/components/crudList.js]: - [x] Please add a method to update all the indexes of the remaining elements every time you delete one or more elements from the array.

**Review by voscarmv — CHANGES_REQUESTED** [review]
Hi @EroAuditore , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

- Please remember to update all indexes every time you remove an element from the array!
- Please add an instruction in your `README.md` file to `npm install` before `npm run build`

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- Adding a Live Demo link to your `README`

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/crudList.js (hunk 1 above)
Comment (sinansevgi) [reply src/components/crudList.js]: :heavy_check_mark: Well Done

**Review by sinansevgi — APPROVED** [review]
# Approved :heavy_check_mark: 

Hi @EroAuditore,

Great work with making all the changes requested by the previous TSE. Indexes getting updated properly now and your readme file is well descriptive. :+1: 

Your project is complete!  
Congratulations! 🎉

You can merge this pull request and continue with your journey :wave: 


Happy Hacking! :computer: 

-----

You can reply to this conversation if you have any questions or need elaboration. :wave: 
Sinan Sevgi


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**